pkg/endpoint/bpf.go

```diff
@@ -542,26 +545,63 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, err
 		}
 	}
 
+	var (
+		consumersAdd, consumersRm policy.SecurityIDContexts
+		policyChanged             bool
+	)
 	// Only generate & populate policy map if a seclabel and consumer model is set up
 	if c != nil {
 		c.AddMap(e.PolicyMap)
 
 		// Regenerate policy and apply any options resulting in the
 		// policy change.
 		// This also populates e.PolicyMap
-		var (
-			consumersToRm, consumersAdd policy.RuleContexts
-			flushEndpointCT             bool
-		)
-		_, flushEndpointCT, consumersAdd, consumersToRm, err = e.regeneratePolicy(owner, nil)
+		policyChanged, consumersAdd, consumersRm, err = e.regeneratePolicy(owner, nil)
 		if err != nil {
 			e.Mutex.Unlock()
-			err = fmt.Errorf("Unable to regenerate policy for '%s': %s",
-				e.PolicyMap.String(), err)
-			return 0, err
+			return 0, fmt.Errorf("Unable to regenerate policy for '%s': %s", e.PolicyMap.String(), err)
+		}
+		// policyChanged can still be true and, at the same time,
+		// the consumersAdd be nil. If this happens it means
+		// the L4-L7 was changed so we need to update the
+		// L3L4Policy map with the new proxyport.
+		if policyChanged &&
+			consumersAdd == nil &&
+			c.L4Policy != nil &&
+			c.L4Policy.Ingress != nil &&
+			c.L3L4Policy != nil {
+
+			// Only update CT if the new a RedirectPort was changed.
+			policyChanged = false
```

Comment:
Do we need some sort of locking for `policyChanged` writes/reads? Can there be concurrent access?